File: rest-service/src/main/java/mb/lib/query/db/SelectChildBlastLinks.java
package mb.lib.query.db;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.util.List;

import io.vulpine.lib.query.util.basic.BasicPreparedListReadQuery;
import mb.lib.model.HashID;
import mb.lib.query.model.BlastJobLink;

public class SelectChildBlastLinks
{
  private static final String Query = """
    SELECT
      *
    FROM
      userlogins5.multiblast_job_to_jobs
    WHERE
      parent_digest = ?
    ORDER BY
      position ASC
    """;

  private final Connection con;
  private final HashID     parentJobID;

  public SelectChildBlastLinks(Connection con, HashID parentJobID) {
    this.con         = con;
    this.parentJobID = parentJobID;
  }

  public List<BlastJobLink> run() throws Exception {
    return new BasicPreparedListReadQuery<>(Query, con, Util::parseJobLink, this::prep)
      .execute()
      .getValue();
  }

  private void prep(PreparedStatement ps) throws Exception {
    ps.setBytes(1, parentJobID.bytes());
  }
}

File: rest-service/src/main/java/mb/lib/query/db/SelectChildBlastLinks.kt
package mb.lib.query.db

import io.vulpine.lib.query.util.basic.BasicPreparedListReadQuery
import mb.lib.model.HashID
import mb.lib.query.model.BlastJobLink
import java.sql.Connection
import java.sql.PreparedStatement

data class SelectChildBlastLinks(private val con: Connection, private val parentJobID: HashID) {
  companion object {
    private const val Query = """
    SELECT
      *
    FROM
      userlogins5.multiblast_job_to_jobs
    WHERE
      parent_digest = ?
    ORDER BY
      position ASC
    """
  }

  fun run(): List<BlastJobLink> = BasicPreparedListReadQuery(Query, con, ::parseJobLink, this::prep).execute().value

  private fun prep(ps: PreparedStatement) = ps.setBytes(1, parentJobID.bytes)
}
